### PR TITLE
luaPackages: update on 2024-06-24

### PR DESCRIPTION
**Pull request by teto**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by teto** [comment]
Result of `nixpkgs-review pr 322274` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>31 packages marked as broken and skipped:</summary>
  <ul>
    <li>lua51Packages.lua-resty-jwt</li>
    <li>lua51Packages.lua-resty-openidc</li>
    <li>lua52Packages.gitsigns-nvim</li>
    <li>lua52Packages.lua-resty-jwt</li>
    <li>lua52Packages.lua-resty-openidc</li>
    <li>lua52Packages.neotest</li>
    <li>lua52Packages.nfd</li>
    <li>lua52Packages.telescope-manix</li>
    <li>lua52Packages.telescope-nvim</li>
    <li>lua53Packages.gitsigns-nvim</li>
    <li>lua53Packages.lua-resty-jwt</li>
    <li>lua53Packages.lua-resty-openidc</li>
    <li>lua53Packages.neotest</li>
    <li>lua53Packages.nfd</li>
    <li>lua53Packages.telescope-manix</li>
    <li>lua53Packages.telescope-nvim</li>
    <li>lua54Packages.gitsigns-nvim</li>
    <li>lua54Packages.ldoc</li>
    <li>lua54Packages.lua-lsp</li>
    <li>lua54Packages.lua-resty-jwt</li>
    <li>lua54Packages.lua-resty-openidc</li>
    <li>lua54Packages.neotest</li>
    <li>lua54Packages.nfd</li>
    <li>lua54Packages.nvim-cmp</li>
    <li>lua54Packages.plenary-nvim</li>
    <li>lua54Packages.sqlite</li>
    <li>lua54Packages.telescope-manix</li>
    <li>lua54Packages.telescope-nvim</li>
    <li>luajitPackages.lua-resty-jwt</li>
    <li>luajitPackages.lua-resty-openidc</li>
    <li>luajitPackages.neotest</li>
  </ul>
</details>
<details>
  <summary>262 packages built:</summary>
  <ul>
    <li>awesome</li>
    <li>awesome.doc</li>
    <li>lua51Packages.argparse</li>
    <li>lua51Packages.busted</li>
    <li>lua51Packages.cassowary</li>
    <li>lua51Packages.cldr</li>
    <li>lua51Packages.commons-nvim</li>
    <li>lua51Packages.cyan</li>
    <li>lua51Packages.dkjson</li>
    <li>lua51Packages.fidget-nvim</li>
    <li>lua51Packages.fluent</li>
    <li>lua51Packages.fzf-lua</li>
    <li>lua51Packages.fzy</li>
    <li>lua51Packages.gitsigns-nvim</li>
    <li>lua51Packages.haskell-tools-nvim</li>
    <li>lua51Packages.image-nvim</li>
    <li>lua51Packages.ldoc</li>
    <li>lua51Packages.lua-lsp</li>
    <li>lua51Packages.lua-protobuf</li>
    <li>lua51Packages.lua-resty-openssl</li>
    <li>lua51Packages.lua-resty-session</li>
    <li>lua51Packages.luacheck</li>
    <li>lua51Packages.luaprompt</li>
    <li>lua51Packages.luarocks</li>
    <li>lua51Packages.luarocks-build-treesitter-parser</li>
    <li>lua51Packages.luarocksCheckHook</li>
    <li>lua51Packages.luasystem</li>
    <li>lua51Packages.lz-n</li>
    <li>lua51Packages.moonscript</li>
    <li>lua51Packages.neotest</li>
    <li>lua51Packages.nfd</li>
    <li>lua51Packages.nvim-cmp</li>
    <li>lua51Packages.nvim-nio</li>
    <li>lua51Packages.pathlib-nvim</li>
    <li>lua51Packages.penlight</li>
    <li>lua51Packages.plenary-nvim</li>
    <li>lua51Packages.rest-nvim</li>
    <li>lua51Packages.rocks-config-nvim</li>
    <li>lua51Packages.rocks-dev-nvim</li>
    <li>lua51Packages.rocks-git-nvim</li>
    <li>lua51Packages.rocks-nvim</li>
    <li>lua51Packages.rustaceanvim</li>
    <li>lua51Packages.serpent</li>
    <li>lua51Packages.sqlite</li>
    <li>lua51Packages.teal-language-server</li>
    <li>lua51Packages.telescope-manix</li>
    <li>lua51Packages.telescope-nvim</li>
    <li>lua51Packages.tl</li>
    <li>lua51Packages.tree-sitter-norg</li>
    <li>lua51Packages.vusted</li>
    <li>lua51Packages.xml2lua</li>
    <li>lua52Packages.argparse</li>
    <li>lua52Packages.busted</li>
    <li>lua52Packages.cassowary</li>
    <li>lua52Packages.cldr</li>
    <li>lua52Packages.commons-nvim</li>
    <li>lua52Packages.cyan</li>
    <li>lua52Packages.dkjson</li>
    <li>lua52Packages.fidget-nvim</li>
    <li>lua52Packages.fluent</li>
    <li>lua52Packages.fzf-lua</li>
    <li>lua52Packages.fzy</li>
    <li>lua52Packages.haskell-tools-nvim</li>
    <li>lua52Packages.image-nvim</li>
    <li>lua52Packages.ldoc</li>
    <li>lua52Packages.lua-lsp</li>
    <li>lua52Packages.lua-protobuf</li>
    <li>lua52Packages.lua-resty-openssl</li>
    <li>lua52Packages.lua-resty-session</li>
    <li>lua52Packages.luacheck</li>
    <li>lua52Packages.luaprompt</li>
    <li>lua52Packages.luarocks</li>
    <li>lua52Packages.luarocks-build-treesitter-parser</li>
    <li>lua52Packages.luarocksCheckHook</li>
    <li>lua52Packages.luasystem</li>
    <li>lua52Packages.moonscript</li>
    <li>lua52Packages.nvim-cmp</li>
    <li>lua52Packages.nvim-nio</li>
    <li>lua52Packages.pathlib-nvim</li>
    <li>lua52Packages.penlight</li>
    <li>lua52Packages.plenary-nvim</li>
    <li>lua52Packages.rest-nvim</li>
    <li>lua52Packages.rocks-config-nvim</li>
    <li>lua52Packages.rocks-dev-nvim</li>
    <li>lua52Packages.rocks-git-nvim</li>
    <li>lua52Packages.rocks-nvim</li>
    <li>lua52Packages.rustaceanvim</li>
    <li>lua52Packages.serpent</li>
    <li>lua52Packages.sqlite</li>
    <li>lua52Packages.teal-language-server</li>
    <li>lua52Packages.tl</li>
    <li>lua52Packages.tree-sitter-norg</li>
    <li>lua52Packages.vusted</li>
    <li>lua52Packages.xml2lua</li>
    <li>lua53Packages.argparse</li>
    <li>lua53Packages.busted</li>
    <li>lua53Packages.cassowary</li>
    <li>lua53Packages.cldr</li>
    <li>lua53Packages.commons-nvim</li>
    <li>lua53Packages.cyan</li>
    <li>lua53Packages.dkjson</li>
    <li>lua53Packages.fidget-nvim</li>
    <li>lua53Packages.fluent</li>
    <li>lua53Packages.fzf-lua</li>
    <li>lua53Packages.fzy</li>
    <li>lua53Packages.haskell-tools-nvim</li>
    <li>lua53Packages.image-nvim</li>
    <li>lua53Packages.ldoc</li>
    <li>lua53Packages.lua-lsp</li>
    <li>lua53Packages.lua-protobuf</li>
    <li>lua53Packages.lua-resty-openssl</li>
    <li>lua53Packages.lua-resty-session</li>
    <li>lua53Packages.luacheck</li>
    <li>lua53Packages.luaprompt</li>
    <li>lua53Packages.luarocks</li>
    <li>lua53Packages.luarocks-build-treesitter-parser</li>
    <li>lua53Packages.luarocksCheckHook</li>
    <li>lua53Packages.luasystem</li>
    <li>lua53Packages.moonscript</li>
    <li>lua53Packages.nvim-cmp</li>
    <li>lua53Packages.nvim-nio</li>
    <li>lua53Packages.pathlib-nvim</li>
    <li>lua53Packages.penlight</li>
    <li>lua53Packages.plenary-nvim</li>
    <li>lua53Packages.rest-nvim</li>
    <li>lua53Packages.rocks-config-nvim</li>
    <li>lua53Packages.rocks-dev-nvim</li>
    <li>lua53Packages.rocks-git-nvim</li>
    <li>lua53Packages.rocks-nvim</li>
    <li>lua53Packages.rustaceanvim</li>
    <li>lua53Packages.serpent</li>
    <li>lua53Packages.sqlite</li>
    <li>lua53Packages.teal-language-server</li>
    <li>lua53Packages.tl</li>
    <li>lua53Packages.tree-sitter-norg</li>
    <li>lua53Packages.vusted</li>
    <li>lua53Packages.xml2lua</li>
    <li>lua54Packages.argparse</li>
    <li>lua54Packages.busted</li>
    <li>lua54Packages.cassowary</li>
    <li>lua54Packages.cldr</li>
    <li>lua54Packages.commons-nvim</li>
    <li>lua54Packages.cyan</li>
    <li>lua54Packages.dkjson</li>
    <li>lua54Packages.fidget-nvim</li>
    <li>lua54Packages.fluent</li>
    <li>lua54Packages.fzf-lua</li>
    <li>lua54Packages.fzy</li>
    <li>lua54Packages.haskell-tools-nvim</li>
    <li>lua54Packages.image-nvim</li>
    <li>lua54Packages.lua-protobuf</li>
    <li>lua54Packages.lua-resty-openssl</li>
    <li>lua54Packages.lua-resty-session</li>
    <li>lua54Packages.luacheck</li>
    <li>lua54Packages.luaprompt</li>
    <li>lua54Packages.luarocks</li>
    <li>lua54Packages.luarocks-build-treesitter-parser</li>
    <li>lua54Packages.luarocksCheckHook</li>
    <li>lua54Packages.luasystem</li>
    <li>lua54Packages.moonscript</li>
    <li>lua54Packages.nvim-nio</li>
    <li>lua54Packages.pathlib-nvim</li>
    <li>lua54Packages.penlight</li>
    <li>lua54Packages.rest-nvim</li>
    <li>lua54Packages.rocks-config-nvim</li>
    <li>lua54Packages.rocks-dev-nvim</li>
    <li>lua54Packages.rocks-git-nvim</li>
    <li>lua54Packages.rocks-nvim</li>
    <li>lua54Packages.rustaceanvim</li>
    <li>lua54Packages.serpent</li>
    <li>lua54Packages.teal-language-server</li>
    <li>lua54Packages.tl</li>
    <li>lua54Packages.tree-sitter-norg</li>
    <li>lua54Packages.vusted</li>
    <li>lua54Packages.xml2lua</li>
    <li>luajitPackages.argparse</li>
    <li>luajitPackages.busted</li>
    <li>luajitPackages.cassowary</li>
    <li>luajitPackages.cldr</li>
    <li>luajitPackages.commons-nvim</li>
    <li>luajitPackages.cyan</li>
    <li>luajitPackages.dkjson</li>
    <li>luajitPackages.fidget-nvim</li>
    <li>luajitPackages.fluent</li>
    <li>luajitPackages.fzf-lua</li>
    <li>luajitPackages.fzy</li>
    <li>luajitPackages.gitsigns-nvim</li>
    <li>luajitPackages.haskell-tools-nvim</li>
    <li>luajitPackages.image-nvim</li>
    <li>luajitPackages.ldoc</li>
    <li>luajitPackages.lua-lsp</li>
    <li>luajitPackages.lua-protobuf</li>
    <li>luajitPackages.lua-resty-openssl</li>
    <li>luajitPackages.lua-resty-session</li>
    <li>luajitPackages.luacheck</li>
    <li>luajitPackages.luaprompt</li>
    <li>luajitPackages.luarocks</li>
    <li>luajitPackages.luarocks-build-treesitter-parser</li>
    <li>luajitPackages.luarocksCheckHook</li>
    <li>luajitPackages.luasystem</li>
    <li>luajitPackages.lz-n</li>
    <li>luajitPackages.moonscript</li>
    <li>luajitPackages.nfd</li>
    <li>luajitPackages.nvim-cmp</li>
    <li>luajitPackages.nvim-nio</li>
    <li>luajitPackages.pathlib-nvim</li>
    <li>luajitPackages.penlight</li>
    <li>luajitPackages.plenary-nvim</li>
    <li>luajitPackages.rest-nvim</li>
    <li>luajitPackages.rocks-config-nvim</li>
    <li>luajitPackages.rocks-dev-nvim</li>
    <li>luajitPackages.rocks-git-nvim</li>
    <li>luajitPackages.rocks-nvim</li>
    <li>luajitPackages.rustaceanvim</li>
    <li>luajitPackages.serpent</li>
    <li>luajitPackages.sqlite</li>
    <li>luajitPackages.teal-language-server</li>
    <li>luajitPackages.telescope-manix</li>
    <li>luajitPackages.telescope-nvim</li>
    <li>luajitPackages.tl</li>
    <li>luajitPackages.tree-sitter-norg</li>
    <li>luajitPackages.vusted</li>
    <li>luajitPackages.xml2lua</li>
    <li>mpvScripts.mpv-webm</li>
    <li>nelua</li>
    <li>nvimpager</li>
    <li>rehex</li>
    <li>sile</li>
    <li>sile.dev</li>
    <li>sile.doc</li>
    <li>sile.man</li>
    <li>vimPlugins.ChatGPT-nvim</li>
    <li>vimPlugins.cmp-fuzzy-buffer</li>
    <li>vimPlugins.cmp-fuzzy-path</li>
    <li>vimPlugins.fidget-nvim</li>
    <li>vimPlugins.fuzzy-nvim</li>
    <li>vimPlugins.fzf-lua</li>
    <li>vimPlugins.haskell-tools-nvim</li>
    <li>vimPlugins.image-nvim</li>
    <li>vimPlugins.lz-n</li>
    <li>vimPlugins.neotest</li>
    <li>vimPlugins.neotest-playwright</li>
    <li>vimPlugins.nvim-dap-ui</li>
    <li>vimPlugins.nvim-nio</li>
    <li>vimPlugins.nvim-teal-maker</li>
    <li>vimPlugins.octo-nvim</li>
    <li>vimPlugins.rest-nvim</li>
    <li>vimPlugins.rocks-config-nvim</li>
    <li>vimPlugins.rocks-nvim</li>
    <li>vimPlugins.rustaceanvim</li>
    <li>vimPlugins.telescope-cheat-nvim</li>
    <li>vimPlugins.telescope-frecency-nvim</li>
    <li>vimPlugins.telescope-fzf-native-nvim</li>
    <li>vimPlugins.telescope-fzf-writer-nvim</li>
    <li>vimPlugins.telescope-fzy-native-nvim</li>
    <li>vimPlugins.telescope-manix</li>
    <li>vimPlugins.telescope-media-files-nvim</li>
    <li>vimPlugins.telescope-nvim</li>
    <li>vimPlugins.telescope-symbols-nvim</li>
    <li>vimPlugins.telescope-undo-nvim</li>
    <li>vimPlugins.telescope-z-nvim</li>
    <li>vimPlugins.telescope-zoxide</li>
  </ul>
</details>

**Comment by teto** [comment]
waiting for staging-next before merging to avoid conflicts.

File: pkgs/development/lua-modules/aliases.nix [hunk 1]
```diff
@@ -43,4 +43,5 @@ mapAliases {
   cyrussasl = throw "cyrussasl was removed because broken and unmaintained "; # added 2023-10-18
   nlua-nvim = throw "nlua-nvim was removed, use neodev-nvim instead"; # added 2023-12-16
   nvim-client = throw "nvim-client was removed because it is now part of neovim"; # added 2023-12-17
+  toml = throw "toml was removed because unused. Feel free to restore"; # added 2024-06-25
```
Comment (mrcjkb) [inline pkgs/development/lua-modules/aliases.nix:46]: suggestion: add a note that toml-edit can also be used

**Review by mrcjkb — APPROVED** [review]

File: pkgs/development/lua-modules/aliases.nix (hunk 1 above)
Comment (teto) [reply pkgs/development/lua-modules/aliases.nix]: done

**Review by mrcjkb — APPROVED** [review]

**Pull request merged**